src/miner.cpp

```diff
@@ -39,13 +39,21 @@ int64_t UpdateTime(CBlockHeader* pblock, const Consensus::Params& consensusParam
     return nNewTime - nOldTime;
 }
 
-void RegenerateCommitments(CBlock& block, CBlockIndex* prev_block)
+void RegenerateCommitments(CBlock& block, ChainstateManager& chainman)
 {
     CMutableTransaction tx{*block.vtx.at(0)};
     tx.vout.erase(tx.vout.begin() + GetWitnessCommitmentIndex(block));
     block.vtx.at(0) = MakeTransactionRef(tx);
 
-    WITH_LOCK(::cs_main, assert(g_chainman.m_blockman.LookupBlockIndex(block.hashPrevBlock) == prev_block));
+    CBlockIndex* prev_block;
+    {
+        // TODO: Temporary scope to check correctness of refactored code.
+        // Should be removed manually after merge of
+        // https://github.com/bitcoin/bitcoin/pull/20158
+        LOCK(::cs_main);
+        assert(std::addressof(g_chainman.m_blockman) == std::addressof(chainman.m_blockman));
+        prev_block = chainman.m_blockman.LookupBlockIndex(block.hashPrevBlock);
```

Comment:
Hm, shouldn't we be asserting here (instead of assigning) to preserve behavior?

Reply:
So `::RegenerateCommitments` has been changed a few times now:
1. In 2c3ba00693, where we passed in a `BlockManager`, instead of reaching for `g_chainman.m_blockman`
    - Because we don't want to reach for globals anymore (the overall point of this PR series)
2. In cced0f46c9, where we passed in the specific `CBlockIndex` instead of a `BlockManager`
    - Because it is a more specific validation-related object than `BlockManager`
3. In e6b4aa6eb5, where we pass in a `ChainstateManger` instead of the specific `CBlockIndex`
    - Because then we don't have to worry about callers violating the invariant checked here: https://github.com/bitcoin/bitcoin/blob/cced0f46c9133e0fc6211e987421ad1d9be1a399/src/miner.cpp#L48 

Just to be clear, the original behaviour does not do any checking/assertions:
https://github.com/bitcoin/bitcoin/blob/2afcf24408b4453e4418ebfb326b141f6ea8647c/src/miner.cpp#L42-L51

So overall, I don't think an behaviour has changed, and I think we should just go with the current one since the few versions we've tried are all functionally equivalent.